### PR TITLE
retyer mechanism

### PR DESCRIPTION
**Pull request by lior-gershony**
the purpose of this PR is to set a retry mechanism for error handling logic.
please note that currently the retyer is set with dummy configuration (exceptions, timeouts etc...) - in the next PR i'll set the actual configuration according to FB API and the required logic. the idea behind this PR is to set the base mechanism for such logic and some flow adjustments to support failure response with an error message.

SOCIAL-36389

### CODE REVIEW

**Comment by morantavori** [comment]
@lior-gershony mind adding in a comment an example response? It'll make reviewing much easier :pray:

**Comment by lior-gershony** [comment]
@morantavori sure.
**example response from Facebook:**
{"data":[{"name":"1545831573852_DPAM","id":"6119405135452"},{"name":"1545831174553_DCO","id":"6119404806252"},{"name":"SK-BM-upload_PPLA","id":"6119402795852"},{"name":"1545826858371_DLO","id":"6119401638652"},{"name":"1545826845938_DLO","id":"6119401357852"},{"name":"1545823174079_DLO","id":"6119398131652"},{"name":"1545822483488_DLO","id":"6119397918452"},{"name":"DPA","id":"6119393658852"},{"name":"Brand awareness","id":"6119392307252"},{"name":"1545812670740_PPPA","id":"6119391117252"},{"name":"Page post link ad","id":"6119385640252"},{"name":"APP_MAI_AIP","id":"6119370383852"},{"name":"DPA-US-18-65-DAT","id":"6119370379852"},{"name":"Daily","id":"6119370378652"},{"name":"Daily","id":"6119370376052"},{"name":"Daily","id":"6119370375452"},{"name":"Daily","id":"6119370374252"},{"name":"Daily","id":"6119370373852"},{"name":"Daily","id":"6119370373252"},{"name":"DPA-US-18-65-DATM","id":"6119370372652"},{"name":"Daily","id":"6119370371652"},{"name":"Campaign_Name","id":"6119370371252"},{"name":"Daily","id":"6119370370652"},{"name":"DPA-US-18-65-DAT","id":"6119370369452"},{"name":"Campaign_Name","id":"6119370366452"}],"paging":{"cursors":{"before":"NjExOTQwNTEzNTQ1MgZDZD","after":"NjExOTM3MDM2NjQ1MgZDZD"},"next":"https:\/\/graph.facebook.com\/v3.1\/act_359773042\/campaigns?access_token=EAAAAIm3moD0BAKrX0FZCY5BeMHrWIm9kDYFkq0Y3ZAElMlNQhZCGkoVfFErQYM5DN1suES57w6Ijs0XcB39SVs7oZCrLVSWv083oh97zEWZBZAHzkovfBL2kYj2cF93jORxLBimcEq6YVUcadqNbZBl8BFNT8v5snXChzxhG0s28c2w3tteoT7RoZBhMXPQqZB6pQchYRNLFZAvAZDZD&fields=name&limit=25&after=NjExOTM3MDM2NjQ1MgZDZD"}}

**example response from the service:**
{"campaignROs":[{"id":6119332169652,"name":"CBO","status":"Active"},{"id":6119313596052,"name":"Daily","status":"Active"},{"id":6119313594852,"name":"Daily","status":"Active"}],"errorMessage":""}

* in case of an error, the `campaignROs` will be empty and the `errorMessage` will contain the error description.

**Comment by morantavori** [comment]
As discussed offline - @lior-gershony is making a few changes to the API between the facade and the service (i.e. changes to the entities passed between them), so I'm holding off my review until that's in :)

**Comment by lior-gershony** [comment]
@morantavori thanks a lot for the review and the guidance!
pushed a fix

File: facade/src/main/java/com/kenshoo/facebookgateway/facades/FacebookFacade.java [hunk 1]
```diff
@@ -15,17 +21,32 @@
...
+        } catch (ExecutionException e) {
+            throw new RuntimeException("internal error during execution: " + e.getMessage());
+        } catch (RetryException e) {
+            throw new RuntimeException(e.getLastFailedAttempt().getExceptionCause().getMessage());
+        }
+    }
+
+    private CampaignsWrapperRO getCampaignsWrapperRO(final String uri) {
+        final Invocation.Builder response = jerseyClient.target(UriBuilder.fromUri(uri).build()).request();
+        final int responseStatusCode = response.head().getStatus();
```
Comment (morantavori) [inline facade/src/main/java/com/kenshoo/facebookgateway/facades/FacebookFacade.java]: As discussed, I'm waiting for input on our attempt to fix this double call

**Comment by lior-gershony** [comment]
retest this please

**Comment by lior-gershony** [comment]
@morantavori pushed another fix

**Pull request merged**